src/hist/basehist.py

```diff
@@ -48,19 +52,28 @@ def __call__(self, *args, **kwargs):
 class BaseHist(bh.Histogram):
     __slots__ = ("_ax", "_storage_proxy")
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args, storage: Optional[Storage] = None, metadata=None):
         """
             Initialize BaseHist object. Axis params can contain the names.
         """
         # TODO: Make a base class type Axis for Hist
         self._ax: List[bh.axis.Axis] = []
         self._hist: Any = None
-        self._storage_proxy: Any = None
+        self._storage_proxy: Optional[Storage] = None
         self.axes: NamedAxesTuple
 
         if len(args):
             self._hist = None
             self._ax = []
+            kwargs = {}  # Only needed for boost-histogram < 0.10.0
```

Comment:
I've fixed this requirement upstream, we should be able to do `metadata=metadata, storage=storage` soon.